Game/model/model.gd

```diff
@@ -79,3 +95,53 @@ func set_ore_at(player_id: int, x: int, y: int, ore: Types.Ore) -> void:
 		ores_layout_updated.emit()
 	else:
 		print("trying to write ore to factory layer: (%d, %d, %d, %s)" % [player_id, x, y, ore])
+
+
+## Get the building type at the given position.
+func get_building_at(pos: TileMapPosition) -> Types.Building:
```

Comment:
I like the use of PlacedBuilding and TileMapPosition to do this. It makes for a really good separation between the *data* of the building (BuildingResource) and an *instance* of a building (PlacedBuilding)  Good job!